### PR TITLE
Add etcd test

### PR DESCRIPTION
**Pull request by ssyssy**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -67,30 +84,93 @@ def docker_alluxio():
...
     host = parsed_url.hostname
     port = parsed_url.port
     alluxio_client = AlluxioClient(worker_hosts=host, worker_http_port=port)
     yield alluxio_client
+
+
+@pytest.fixture
+def etcd_alluxio_client(docker_alluxio_with_etcd):
+    LOGGER.debug(
+        f"get etcd AlluxioFileSystem connect to {docker_alluxio_with_etcd}"
```
Comment (LuQQiu) [inline tests/conftest.py:171]: AlluxioFileSystem -> AlluxioClient

File: tests/conftest.py [hunk 2]
```diff
@@ -67,30 +84,93 @@ def docker_alluxio():
...
-                raise SystemError from e
-            time.sleep(10)
+    yield yield_url(url)
     stop_docker(worker_container)
     stop_docker(master_container)
 
 
+@pytest.fixture(scope="module")
+def docker_alluxio_with_etcd():
+    if "ALLUXIO_URL" in os.environ:
```
Comment (LuQQiu) [inline tests/conftest.py:94]: Can you merge the docker_alluxio() method and docker_alluxio_with_etcd() method to a method like launch_alluxio(etcd_hosts)
Most of the logics are shared

File: tests/conftest.py [hunk 3]
```diff
@@ -26,14 +29,25 @@ def stop_docker(container):
...
+                return url
+                break
+        except Exception as e:  # noqa: E722
+            timeout -= 1
+            if timeout < 0:
+                raise SystemError from e
+            time.sleep(10)
+
+
+def launch_alluxio_dockers(is_etcd=False):
```
Comment (LuQQiu) [inline tests/conftest.py:50]: is_etcd => with_etcd?

File: tests/conftest.py [hunk 4]
```diff
@@ -46,51 +60,102 @@ def docker_alluxio():
...
+            "-Dalluxio.etcd.endpoints=http://etcd:2379 "
+            if is_etcd
+            else ""
+        )
+        + '-Dalluxio.underfs.xattr.change.enabled=false " alluxio/alluxio:310-SNAPSHOT master'
     )
+
     run_cmd_worker = (
-        "docker run --platform linux/amd64 -d --rm --net=alluxio_network -p 28080:28080 -p 29999:29999 -p 29997:29997 "
+        "docker run --platform linux/amd64 -d --rm -h localhost --net=alluxio_network -p 28080:28080 -p 29999:29999 -p 29997:29997 "
```
Comment (LuQQiu) [inline tests/conftest.py:73]: why -h localhost is needed for worker but not for master?

**Review by LuQQiu — COMMENTED** [review]
Cool, left some minor comments

File: tests/conftest.py (hunk 4 above)
Comment (LuQQiu) [reply tests/conftest.py:73]: why need localhost parameter?

Comment (ssyssy) [reply tests/conftest.py:73]: So when worker register itself into etcd? It uses its own hostname to register. That hostname will be its container hash without specifying the hostname here. So the python client will not get the correct worker host from etcd.

Comment (LuQQiu) [reply tests/conftest.py:73]: @ssz1997 is it the recommended way?

Comment (ssyssy) [reply tests/conftest.py]: @LuQQiu Tried another way, use alluxio config to do that.

**Review by LuQQiu — APPROVED** [review]
LGTM！

**Pull request merged**